patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
```

Comment:
The pattern accomplishes same things as python's `with`, C#'s `using`, Java's `try-with-resources`, C++'s RAII. Here's a example http://promise-nuggets.github.io/articles/21-context-managers-transactions.html